server/command/command.go

```diff
@@ -193,38 +203,39 @@ func newCommand(args *model.CommandArgs, conf *config.Config, a splunk.Splunk) *
 
 // alertSubscriptionMessage creates message for alert subscription
 // returns message text and unique id for alert
-func alertSubscriptionMessage(siteURL string) (string, string) {
+func alertSubscriptionMessage(siteURL, secret string) (string, string) {
 	id := uuid.New()
 	post := fmt.Sprintf(
 		"Added alert\n"+
-			"You can copy following link to your splunk alert action: %s/plugins/%s%s%s?id=%s",
+			"You can copy following link to your splunk alert action: %s/plugins/%s%s%s?id=%s&secret=%s",
 		siteURL,
 		// TODO: Must replace with c.config.PluginID it returns empty string now
 		"com.mattermost.plugin-splunk",
 		config.APIPath,
 		api.WebhookEndpoint,
-		id)
+		id,
+		secret,
+	)
 	return post, id.String()
 }
 
 func (c *command) subscribeAlert(_ ...string) (*model.CommandResponse, error) {
-	message, id := alertSubscriptionMessage(c.args.SiteURL)
-	c.splunk.AddAlertListener(c.args.ChannelId, id, func(payload splunk.AlertActionWHPayload) {
-		_, err := c.splunk.CreatePost(&model.Post{
-			UserId:    c.splunk.BotUser(),
-			ChannelId: c.args.ChannelId,
-			Message:   fmt.Sprintf("New alert action received %s", payload.ResultsLink),
-		})
-		if err != nil {
-			log.Println(err)
-		}
-	})
+	message, id := alertSubscriptionMessage(c.args.SiteURL, c.config.Secret)
+	err := c.splunk.AddAlert(c.args.ChannelId, id)
+	if err != nil {
+		message = err.Error()
```

Comment:
We should log this error right?

Reply:
@mickmister sure , can you check on the message ,if you want some change